### ISSUE TITLE
Support for Vue 3's <script setup>

### ISSUE THREAD

**Issue by tony19** [stackblitz/core#1308]
Vue 3's [`<script setup>`](https://github.com/vuejs/rfcs/blob/sfc-improvements/active-rfcs/0000-sfc-script-setup.md) block is not recognized in Stackblitz, and renders an error message:

```
Script missing
```

Example usage of `<script setup>`:
```html
<template>
  <div>
    <MyButton />
  </div>
</template>

<script setup>
export { default as MyButton } from "./components/MyButton.vue";
</script>
```

[StackBlitz demo](https://stackblitz.com/edit/vue-st7arn)

**Comment by EricSimons** [comment]
Thanks for the repro @tony19! We'll take a look at this and get a fix out asap. Will circle back here when it's live 👍

**Comment by RawZ06** [comment]
Hi, this feature it was beta experimental when this issue was create. 
Today with Vue 3.2 released as the few days, this syntax is integrated in Vue.

But stackblitz render the same error message.

With web container, I advise, to wait, using vite !

**Comment by Bobakanoosh** [comment]
Any updates on this?

**Comment by chrisshroba** [comment]
StackBlitz seems to be the only online code sandbox that supports multiple .vue files (JSFiddle doesn't allow .vue, CodePen supports .vue but only lets you create one file on the free plan, repl.it is kind of a mess to configure for a simple Vue project).

It would be awesome if it could support <script setup>, since that seems to be the future of Vue, and imo the most concise and elegant way to define components. I've been looking for a place to host vue code to demonstrate issues I'm asking about online (e.g. on StackOverflow).

**Comment by Bobakanoosh** [comment]
> StackBlitz seems to be the only online code sandbox that supports multiple .vue files (JSFiddle doesn't allow .vue, CodePen supports .vue but only lets you create one file on the free plan, repl.it is kind of a mess to configure for a simple Vue project).
> 
> It would be awesome if it could support <script setup>, since that seems to be the future of Vue, and imo the most concise and elegant way to define components. I've been looking for a place to host vue code to demonstrate issues I'm asking about online (e.g. on StackOverflow).

https://sfc.vuejs.org/

This does work, however it's pretty simplistic. Definitely meets the "post on stackoverflow" necessity, but not the "use as an online project" necessity.

**Comment by braincomb** [comment]
Would love to have `<script setup>` support, looks like it's still broken :(

**Comment by mmoore99** [comment]
Any updates on this?

**Comment by Theiaz** [comment]
> Any updates on this?

Bump. Had the same problem today. Blocker for me to use stackblitz more frequently. 

Additionally the error message `Script missing` is not that meaningful.

**Comment by Return180bpm** [comment]
Still an issue

**Comment by py-ng** [comment]
Try using this: [vite.new/vue](https://vite.new/vue)

> The official Vue build setup is based on [Vite](https://vitejs.dev/).
> You can try Vue with SFCs online on [StackBlitz](https://vite.new/vue). 
> StackBlitz runs the Vite-based build setup.

Reference: [Vue.js Quickstart | Official Documentation](https://vuejs.org/guide/quick-start.html#online)

**Comment by philpicton** [comment]
This is still an issue. 
If you aren't planning to fix it, could you please maybe consider not presenting a link in your create a new project modal to spin up a Vue3 instance that _literally does not work with the composition api_, and does not offer a useful error message. This would save a lot of time for people like me. Maybe link to the vite.new/vue one mentioned above? 

![image](https://github.com/stackblitz/core/assets/89278080/40390a88-225f-462e-9b8c-3f1626dce4ce)

**Comment by HeyGarrison** [comment]
Hey all - our platform has improved significantly over the past few years. One major improvement is WebContainers and Vite. I think the best solution for this problem is to use vite.new. You can start a new Vue project navigating to https://vite.new/vue.

In general WebContainers is superior to the enviroment running this project in just about every way. If you're curious about the differences between the two environments checkout this link here: https://developer.stackblitz.com/guides/user-guide/available-environments#frontmatter-title

**Comment by jschroeter** [comment]
Yes, using the link you provided solves the issue. But why is the Vue template promoted on Stackblitz still not using the new setup? Fixing this would solve all the confusion I think.

**Comment by HeyGarrison** [comment]
@jschroeter great question!

We actually have that change coming imminently. I think it's literally in a pull request. Initially we really wanted WebContainers to reach a certain maturity point before we switched the starters over. To be honest though, we've been at that point for a little while and are just now making the change. 

All that to say we are very excited to be able to make this change. It'll really help a lot of our users and solve issues like #2792!
